src/main/java/org/mule/extensions/vm/internal/operations/VMOperations.java

```diff
@@ -219,6 +220,22 @@ public void publish(@Content TypedValue<Serializable> content,
 
   private void doPublish(Serializable content, QueueDescriptor queueDescriptor, Queue queue) {
     try {
+      if (queueManager.getQueueConfiguration(queue.getName()).isPersistent()) {
```

Comment:
This should not just be for persistent ones.  The same race condition can occur with in memory queues.